### ISSUE TITLE
Add `--dry-run` option.

### ISSUE THREAD

**Issue by ghuls** [io12/unf#3]
It would be nice to have a `--dry-run` or `-l` (`--list`) option that would just show how files would be renamed, so it is fast to check if `-f` will do the job the user wants.

```bash
$ unf ---dry-run -r My\ Files/ My\ Folder
rename 'My Files/Passwords :) .txt' -> 'My Files/Passwords.txt'
rename 'My Files/Another Cool Photo.JPG' -> 'My Files/Another_Cool_Photo.JPG'
rename 'My Files/Wow Cool Photo.JPG' -> 'My Files/Wow_Cool_Photo.JPG'
rename 'My Files/Cool Photo.JPG' -> 'My Files/Cool_Photo.JPG'
rename 'My Files/' -> 'My_Files'
rename 'My Folder' -> 'My_Folder'
```

**Comment by io12** [comment]
Thanks for the issue. This is a good idea, but the [collision resolution](https://github.com/io12/unf#how-does-this-handle-collisions) might make it hard to implement. For example, if the contents of a directory are `a_001.txt` `a_001)).txt` `a_002))))).txt`, renaming the files will create a collision, and `unf` checks for collisions as it's renaming. To handle `--dry-run`, `unf` would need to scan the files, load them into a virtual file tree, and rename them inside the virtual file tree. The unit tests handle this by creating file trees in a temporary directory, although that's probably not the ideal behavior for `--dry-run`. Probably the solution is to make the renaming only run on virtual filesystems, but have a function to actually apply the renames.

**Comment by ghuls** [comment]
Could you not use a HashSet to store all files (and dirs) in a certain directory and modify the HashSet by removing the key with the old filename with an old name and add a new tilename with the new key while checking that that file does not already exist in the HashSet. For the real file renaming, the old code should be used (as new files can appear while renaming others).

**Comment by io12** [comment]
> Could you not use a HashSet to store all files [...]

I think there are cases where the ordering of the files changes what renames happen, and `read_dir()` probably won't return filenames in the same order as `HashSet::iter()`. Probably some other datatype like a `Vec` or `BTreeSet` and enforcing sorting would work, though.

> For the real file renaming, the old code should be used (as new files can appear while renaming others).

Yeah, this is a good point.

**Comment by io12** [comment]
Fixed in 2.1.0. I ended up using an [in-memory filesystem library](https://docs.rs/rsfs/0.4.1/rsfs/mem/index.html).
